### PR TITLE
Add exampls 

### PR DESCRIPTION
**Pull request by tarekbadrshalaan**
this pull request related to this issue https://github.com/segmentio/kafka-go/issues/210
I add four example/use-cases how to use **kafka-go**
-  **consumer-logger** : simple example, logs the result consumed from the topic.
- **consumer-mongo-db** : example with mongo-db, insert the the topic data into mongo-db.
- **producer-random** : simple example of how to create random UUID generator and insert the result into kafka topic.
- **producer-api** : example with an API end point push the requests data into Kafka topic.

to make every thing work together.
there is **docker-compose** file manage the communication between the container.
additional images : 
- zookeeper
- mongo-db
- kafka 

please note : the file **kafka/kafka-variables.env** it's just for Kafka conflagration we can use anther setup if we need.

### CODE REVIEW

File: examples/producer-api/main.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		Topic:    topic,
+		Balancer: &kafka.LeastBytes{},
+	})
+}
+
+func main() {
+	// get kafka writer using environment variables.
+	kafkaURL := os.Getenv("kafkaURL")
+	topic := os.Getenv("topic")
+	kafkaWriter := getKafkaWriter(kafkaURL, topic)
```
Comment (NeMO84) [inline examples/producer-api/main.go]: defer close the writer?

File: examples/producer-random/main.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			Value: []byte(fmt.Sprint(uuid.New())),
+		}
+		err := writer.WriteMessages(context.Background(), msg)
+		if err != nil {
+			fmt.Println(err)
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	writer.Close()
```
Comment (NeMO84) [inline examples/producer-random/main.go]: unreachable code, defer close here too?

**Comment by tarekbadrshalaan** [comment]
thanks @NeMO84 for your comments, as you can find in the last commit I solved unreachable code by adding defer close() to reader and writer.

File: examples/producer-random/main.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	kafka "github.com/segmentio/kafka-go"
+)
+
+func getKafkaWriter(kafkaURL, topic string) *kafka.Writer {
```
Comment (achille-roussel) [inline examples/producer-random/main.go]:
```suggestion
func newKafkaWriter(kafkaURL, topic string) *kafka.Writer {
```
It looks like this is more of a constructor than a getter.

File: examples/producer-random/main.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		Topic:    topic,
+		Balancer: &kafka.LeastBytes{},
+	})
+}
+
+func main() {
+	// get kafka writer using environment variables.
+	kafkaURL := os.Getenv("kafkaURL")
+	topic := os.Getenv("topic")
+	writer := getKafkaWriter(kafkaURL, topic)
```
Comment (tarekbadrshalaan) [inline examples/producer-random/main.go]: writer := newKafkaWriter(kafkaURL, topic)

**Comment by achille-roussel** [comment]
Alright, this looks like a good base I think, we can merge and add more as needed.

Thanks for your contribution!

**Pull request merged**